### ISSUE TITLE
When switching languages, the data reactive is lost

### ISSUE THREAD

**Issue by jiaozhiye** [intlify/vue-i18n#753]
### Reporting a bug?

"vue-i18n": "9.2.0-beta.13"
When switching languages, the data reactive is lost, 9.2.0-beta.11 version didn't have this problem!

### Expected behavior

I hope I don't lose the responsiveness

### Reproduction

import { createI18n } from 'vue-i18n';

import enDesign from '@jiaozhiye/qm-design/lib/locale/lang/en';
import zhDesign from '@jiaozhiye/qm-design/lib/locale/lang/zh-cn';

import enLocale from './lang/en';
import zhLocale from './lang/zh-cn';

const messages = {
  [enLocale.name]: {
    qm: enDesign.qm,
    app: enLocale.app,
  },
  [zhLocale.name]: {
    qm: zhDesign.qm,
    app: zhLocale.app,
  },
};

const i18n = createI18n({
  locale: 'zh-cn',
  fallbackLocale: 'zh-cn',
  messages,
});

const { t, mergeLocaleMessage } = i18n.global;

const changeLocale = (lang: string): void => {
  i18n.global.locale = lang;
};

export { t, mergeLocaleMessage, changeLocale };
export default i18n;

// ==============

By calling the changeLocale function to achieve language switch, unfortunately, it didn't have the desired effect.

### System Info

```shell
"vue": "^3.2.20",
"vue-i18n": "^9.2.0-beta.13",
```

### Validations

- [X] Read the [Contributing Guidelines](https://github.com/intlify/vue-i18n-next/blob/master/.github/CONTRIBUTING.md)
- [X] Read the [Documentation](https://vue-i18n.intlify.dev/)
- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.
- [X] Check that this is a concrete bug. For Q&A open a [GitHub Discussion](https://github.com/intlify/vue-i18n-next/discussions)

**Comment by zynth17** [comment]
> @jiaozhiye I could not reproduce this issue. Could you give us the minimum repo code, please?

here is the sandbox 
https://codesandbox.io/s/unruffled-pine-lnccl?file=/src/main.js

**Comment by PeterAlfredLee** [comment]
@zynth17 

> here is the sandbox
> https://codesandbox.io/s/unruffled-pine-lnccl?file=/src/main.js

I just checked. Your error has nothing to do with vue-i18n@9.2.0-beta.13.

First, you should keep id.json the same structure as en.json. Like this:
id.json
```json
{
  "intro": {
    "desc": "Untuk panduan dan tata cara mengatur atau mengubah projek ini, mohon cek "
  }
}
```
Second, there is a wrong line in your App.vue file that should be corrected, as follows:
App.vue
```diff
-    <option v-for="locale in availableLocales" :key="locale" value="locale">
+    <option v-for="locale in availableLocales" :key="locale" :value="locale">
```

Here's the modified code: https://codesandbox.io/s/jolly-archimedes-1l4zo

**Comment by zynth17** [comment]
> @zynth17
> 
> > here is the sandbox
> > https://codesandbox.io/s/unruffled-pine-lnccl?file=/src/main.js
> 
> I just checked. Your error has nothing to do with vue-i18n@9.2.0-beta.13.
> 
> First, you should keep id.json the same structure as en.json. Like this: id.json
> 
> ```json
> {
>   "intro": {
>     "desc": "Untuk panduan dan tata cara mengatur atau mengubah projek ini, mohon cek "
>   }
> }
> ```
> 
> Second, there is a wrong line in your App.vue file that should be corrected, as follows: App.vue
> 
> ```diff
> -    <option v-for="locale in availableLocales" :key="locale" value="locale">
> +    <option v-for="locale in availableLocales" :key="locale" :value="locale">
> ```
> 
> Here's the modified code: https://codesandbox.io/s/jolly-archimedes-1l4zo

ah right i do mistakes in that sandbox. and it does weirdly work in the sandbox. but it does not work with my local.

**Comment by zynth17** [comment]
@PeterAlfredLee you can try my repo here 

https://github.com/zynth17/vitailse/tree/vue-i18n-bug

**Comment by kazupon** [comment]
I've just released v9.2.0-beta.14 !
https://github.com/intlify/vue-i18n-next/releases/tag/v9.2.0-beta.14

Please check it!
Thanks all!

**Comment by ue360** [comment]
![image](https://user-images.githubusercontent.com/367891/138246485-2b4386be-09f1-438f-ae26-f69736f4bb0a.png)
